### DISCUSSION TITLE
Doubt about hetero conv

### DISCUSSION THREAD

**Discussion by DogitoErgoSum** [pyg-team/pytorch_geometric#4464] [Q&A] [answered]
This is my toy model 

![image](https://user-images.githubusercontent.com/51344466/163050491-f89f6c99-81f0-42f0-a5de-fb2ec002193e.png)

Now imagine that i want to pass the following through the previous model 
![image](https://user-images.githubusercontent.com/51344466/163050860-ab4efa93-ff20-46de-a197-cfacdba7537f.png)
My doubt is, are both the convolutions on Node Type C performed at the same time or is one done after the other?

**Comment by rusty1s** [comment, marked as answer]
They are both performed at the same time. That is, input features to GNN layers will always correspond to the features of the previous layer, not to intermediate outputs.
